# PART-1

In [1]:
# importing the necessary 
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb
import warnings
warnings.filterwarnings("ignore")
%matplotlib inline

In [2]:
# loading data in notebook
data=pd.read_csv(r"C:\Users\HP\Downloads\Admission_Predict.csv")

In [3]:
data

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
0             1        337          118                  4  4.5   4.5  9.65   
1             2        324          107                  4  4.0   4.5  8.87   
2             3        316          104                  3  3.0   3.5  8.00   
3             4        322          110                  3  3.5   2.5  8.67   
4             5        314          103                  2  2.0   3.0  8.21   
..          ...        ...          ...                ...  ...   ...   ...   
395         396        324          110                  3  3.5   3.5  9.04   
396         397        325          107                  3  3.0   3.5  9.11   
397         398        330          116                  4  5.0   4.5  9.45   
398         399        312          103                  3  3.5   4.0  8.78   
399         400        333          117                  4  5.0   4.0  9.66   

     Research  Chance of Admit   
0           1              0.92  
1           1              0.76  
2           1              0.72  
3           1              0.80  
4           0              0.65  
..        ...               ...  
395         1              0.82  
396         1              0.84  
397         1              0.91  
398         0              0.67  
399         1              0.95  

[400 rows x 9 columns]

In [4]:
# crateing a copy of the data.
data_copy=data.copy()

In [5]:
# sample of the data
data_copy.sample(5)

Serial No.  GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  \
283         284        321          111                  3  2.5   3.0  8.90   
290         291        307          105                  2  2.5   3.0  7.65   
336         337        319          110                  3  3.0   2.5  8.79   
216         217        322          112                  4  4.5   4.5  9.26   
50           51        313           98                  3  2.5   4.5  8.30   

     Research  Chance of Admit   
283         1              0.80  
290         0              0.58  
336         0              0.72  
216         1              0.91  
50          1              0.76

In [6]:
# checking the shape of data
data_copy.shape

(400, 9)

*There are 400 rows and 9 columns in our dataset.

In [7]:
# information about the data
data_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Serial No.         400 non-null    int64  
 1   GRE Score          400 non-null    int64  
 2   TOEFL Score        400 non-null    int64  
 3   University Rating  400 non-null    int64  
 4   SOP                400 non-null    float64
 5   LOR                400 non-null    float64
 6   CGPA               400 non-null    float64
 7   Research           400 non-null    int64  
 8   Chance of Admit    400 non-null    float64
dtypes: float64(4), int64(5)
memory usage: 28.3 KB


* We have numerial type of dataset.

 In which we have 4 columns based on float data type and 5 columns based on int data type.

In [8]:
# dropping the serial no column from the dataset as it is not much important.
data_copy.drop(columns="Serial No.",axis=1,inplace=True)

In [10]:
# cross checking for the dataset
data_copy.head(5)

GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  Research  \
0        337          118                  4  4.5   4.5  9.65         1   
1        324          107                  4  4.0   4.5  8.87         1   
2        316          104                  3  3.0   3.5  8.00         1   
3        322          110                  3  3.5   2.5  8.67         1   
4        314          103                  2  2.0   3.0  8.21         0   

   Chance of Admit   
0              0.92  
1              0.76  
2              0.72  
3              0.80  
4              0.65

## checking and cleaning the particular columns.

### (1). For GRE Score column

In [12]:
data_copy.sample(2)

GRE Score  TOEFL Score  University Rating  SOP  LOR   CGPA  Research  \
205        295           99                  2  2.5   3.0  7.65         0   
383        300          100                  3  3.0   3.5  8.26         0   

     Chance of Admit   
205              0.57  
383              0.62

In [13]:
# checking the info 
data_copy["GRE Score"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 400 entries, 0 to 399
Series name: GRE Score
Non-Null Count  Dtype
--------------  -----
400 non-null    int64
dtypes: int64(1)
memory usage: 3.3 KB


In [14]:
# checking for the null values
data["GRE Score"].isnull().sum()

0

* There is no null values in it.

In [15]:
# checking the unique values in it.
data_copy["GRE Score"].unique()

array([337, 324, 316, 322, 314, 330, 321, 308, 302, 323, 325, 327, 328,
       307, 311, 317, 319, 318, 303, 312, 334, 336, 340, 298, 295, 310,
       300, 338, 331, 320, 299, 304, 313, 332, 326, 329, 339, 309, 315,
       301, 296, 294, 306, 305, 290, 335, 333, 297, 293], dtype=int64)

In [16]:
data_copy["GRE Score"].nunique()

49

* We have 49 unique values in GRE Score column.

In [17]:
# checking for Maximum Score and Minimum Score from the GRE column.
data_copy["GRE Score"].max()

340

In [18]:
data_copy["GRE Score"].min()

290

* We have 340 as maximum score and 290 as minimum score in GRE Score column. 

In [19]:
# Chceking for value counts.
data_copy["GRE Score"].value_counts()

GRE Score
324    19
312    19
316    14
322    14
314    14
325    13
321    13
311    12
317    12
327    12
320    12
326    11
313    11
308    11
323    10
315    10
300     9
329     9
319     9
318     9
340     8
310     8
301     8
299     8
304     8
331     8
307     8
305     8
298     7
334     7
332     6
328     6
306     6
309     6
330     6
336     5
296     5
303     5
302     5
335     4
295     4
338     4
297     4
333     4
339     3
294     2
290     2
337     1
293     1
Name: count, dtype: int64

### (2). For TOEFL Score columns

In [20]:
#checking the info
data_copy["TOEFL Score"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 400 entries, 0 to 399
Series name: TOEFL Score
Non-Null Count  Dtype
--------------  -----
400 non-null    int64
dtypes: int64(1)
memory usage: 3.3 KB


In [21]:
# checking for null values.
data_copy["TOEFL Score"].isnull().sum()

0

* No null values present in it.

In [22]:
# checking for unique values 
data_copy["TOEFL Score"].unique()

array([118, 107, 104, 110, 103, 115, 109, 101, 102, 108, 106, 111, 112,
       105, 114, 116, 119, 120,  98,  93,  99,  97, 117, 113, 100,  95,
        96,  94,  92], dtype=int64)

In [23]:
data_copy["TOEFL Score"].nunique()

29

* There are total 29 unique values in it.

In [26]:
# checking for maximum and minimum score .
data_copy["TOEFL Score"].max()

120

In [27]:
data_copy["TOEFL Score"].min()

92

* We have 120 as maximum score and 92 as minimum score in TOEFL Score.

In [24]:
# counting the score
data_copy["TOEFL Score"].value_counts()

TOEFL Score
110    37
105    28
107    26
104    25
106    24
112    21
100    19
99     18
111    17
108    17
102    17
109    17
103    16
114    15
113    14
101    11
116    11
98      9
118     9
119     9
115     9
120     8
117     7
97      6
96      4
93      2
95      2
94      1
92      1
Name: count, dtype: int64

* We have 8 students who scored maximum score in TOEFL.